python/whylogs/api/writer/whylabs_base.py

```diff
@@ -0,0 +1,256 @@
+
+from whylogs.api.logger.result_set import ProfileResultSet, ViewResultSet
+from whylogs.api.whylabs.session.session_manager import default_init
+from whylogs.api.writer.whylabs_client import WhyLabsClient
+from whylogs.api.writer.writer import Writable, Writer
+from whylogs.core import DatasetProfileView
+from whylogs.core.dataset_profile import DatasetProfile
+from whylogs.core.errors import BadConfigError
+from whylogs.core.view.segmented_dataset_profile_view import SegmentedDatasetProfileView
+from whylogs.migration.uncompound import (
+    FeatureFlags,
+    _uncompound_dataset_profile,
+    _uncompound_metric_feature_flag,
+)
+
+FIVE_MINUTES_IN_SECONDS = 60 * 5
+DAY_IN_SECONDS = 60 * 60 * 24
+FIVE_YEARS_IN_SECONDS = DAY_IN_SECONDS * 365 * 5
+_WHYLABS_SKIP_CONFIG_READ = os.environ.get("_WHYLABS_SKIP_CONFIG_READ") or False
+
+logger = logging.getLogger(__name__)
+
+
+def _check_whylabs_condition_count_uncompound() -> bool:
+    global _WHYLABS_SKIP_CONFIG_READ
+    if _WHYLABS_SKIP_CONFIG_READ:
+        return True
+    whylabs_config_url = (
+        "https://whylabs-public.s3.us-west-2.amazonaws.com/whylogs_config/whylabs_condition_count_disabled"
+    )
+    logger.info(f"checking: {whylabs_config_url}")
+    try:
+        response = requests.head(whylabs_config_url)
+        logger.info(f"checking: {whylabs_config_url}")
+        logger.info(f"headers are: {response.headers} code: {response.status_code}")
+        if response.status_code == 200:
+            logger.info(
+                "found the whylabs condition count disabled file so running uncompound on condition count metrics"
+            )
+            return True
+        elif response.status_code == 404:
+            logger.info("no whylabs condition count disabled so sending condition count metrics as v1.")
+            return False
+        else:
+            logger.info(f"Got response code {response.status_code} but expected 200, so running uncompound")
+    except Exception:
+        pass
+
+    _WHYLABS_SKIP_CONFIG_READ = True
+    return True
+
+
+class WhyLabsWriterBase(Writer):
+    """
+    Base class for WhyLabs writers
+    """
+
+    def __init__(
+        self,
+        org_id: Optional[str] = None,
+        api_key: Optional[str] = None,
+        dataset_id: Optional[str] = None,
+        api_client: Optional[ApiClient] = None,
+        ssl_ca_cert: Optional[str] = None,
+        _timeout_seconds: Optional[float] = None,
+        whylabs_client: Optional[WhyLabsClient] = None,
+    ):
+        self._whylabs_client: WhyLabsClient = whylabs_client or WhyLabsClient(
+            org_id,
+            api_key,
+            dataset_id,
+            api_client,
+            ssl_ca_cert,
+            _timeout_seconds,
+        )
+        session = default_init()  # Force an init if the user didn't do it, it's idempotent
+        config = session.config
+
+        self._reference_profile_name = config.get_whylabs_refernce_profile_name()
```

Comment:
Keeping these in sync with their values in `_whylabs_client` is error-prone. Keep them here and remove them from `WhyLabsClient`. Just pass them in to the methods that require them.